llvm/lib/SYCLLowerIR/ModuleSplitter.cpp

```diff
@@ -52,9 +52,9 @@ constexpr char SYCL_SCOPE_NAME[] = "<SYCL>";
 constexpr char ESIMD_SCOPE_NAME[] = "<ESIMD>";
 constexpr char ESIMD_MARKER_MD[] = "sycl_explicit_simd";
 
-cl::opt<bool> SupportDynamicLinking{
-    "support-dynamic-linking",
-    cl::desc("Generate device images that are suitable for dynamic linking"),
+cl::opt<bool> AllowDeviceDependencies{
+    "allow-device-dependencies",
```

Comment:
based on the description of the help option, the name `allow-device-image-dependencies` is more clear to me, `allow-device-dependencies` isn't too clear.

Reply:
The clang level option to control this behavior is `-fsycl-allow-device-dependencies`.  Should we be consistent with the clang option or make the option name clearer?

Reply:
The clang option name is unclear to me too, but if we already went through the options working group and they approved it then it's fine to match it here.

Reply:
> Should we be consistent with the clang option or make the option name clearer?

I don't think that we have to 100% match here. `device-dependencies` does not look entirely clear to me. I won't strongly object, but my preference is to have clearer name, even if it is different from `clang` one. 

I would rather talk to options wg once again to change public name as well to `device-image-dependencies` or `device-code-dependencies`

Reply:
I'll ask the options WG.

Reply:
Should we try to change the option name for 2025.0, since it is a new option, or add a synonymous option name `-fsycl-allow-device-image-dependencies` for 2025.1 ?

Reply:
Definitely too late for 2025.0, we would have to deprecate and add the new option moving forward with 2025.1